framework/src/problems/FEProblemBase.C

```diff
@@ -139,6 +139,11 @@ validParams<FEProblemBase>()
                         false,
                         "True to skip additional data in equation system for restart. It is useful "
                         "for starting a transient calculation with a steady-state solution");
+  params.addParam<bool>(
+      "reset_time_after_restart",
```

Comment:
It makes a bit more sense to me for this to be in the Executioner block.

Reply:
I thought about this. If we did not reset time right after it is restarted, we could potentially create inconsistency on the initial evaluations, etc. I initially thought I can make this parameter in `TransientMultiApp`, but that makes things even more delayed and could cause subtle bugs.